ethstorage/miner/l1_mining_api.go

```diff
@@ -142,6 +135,34 @@ func (m *l1MiningAPI) SubmitMinedResult(ctx context.Context, contract common.Add
 		return common.Hash{}, fmt.Errorf("failed to estimate gas: %w", err)
 	}
 	m.lg.Info("Estimated gas done", "gas", estimatedGas)
+	cost := new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasPrice)
```

Comment:
I think we can change the strategy: if mining info changes, start estimating which block we can start to have profit, then start mining from that block.

Reply:
Maybe for every new block, we can first estimate the profit before beginning to mine, and if it doesn't meet the requirement, then we can skip this block? But it also depends on the dynamic gas price when we are estimating.

Reply:
Tracked by https://github.com/ethstorage/es-node/issues/151

Reply:
resolve as will be updated later (https://github.com/ethstorage/es-node/issues/151)